api/src/main/java/com/messagebird/MessageBirdServiceImpl.java

```diff
@@ -485,22 +509,40 @@ private boolean isURLAbsolute(String url) {
             DateFormat df = getDateFormat();
             mapper.setDateFormat(df);
 
-            final String json = mapper.writeValueAsString(body);
-            connection.getOutputStream().write(json.getBytes(String.valueOf(StandardCharsets.UTF_8)));
+            setAdditionalHeaders(connection, headers);
```

Comment:
i placed it in each if-else statement because if they are set above then they could be overwritten by headers that are set inside the if-else statements (e.g. "Content-Type"). therefore, they need to be set after all the other headers in the if-else statements. initially, i tried to set the additional headers once _after_ the if-else block, but this fails when the request is a POST because the connection stream is opened inside the if block, and no more headers can be added to the request once this happens

Reply:
yes you are right about "Content-Type".